### PR TITLE
issue #25 - create, delete calendar event for conference 

### PR DESCRIPTION
**Pull request by ihemantd**
Please review the commit.
Thanks,

### CODE REVIEW

**Comment by raysrashmi** [comment]
I was trying hound on this. thats why closed and re opend this PR :)

**Comment by raysrashmi** [comment]
Can you rebase branch with latest master? We have added hound.yml

**Comment by ihemantd** [comment]
@arunagw @raysrashmi 
How to remove this violation   `trailing Whitespace Violation: Line #179 should have no trailing whitespace ?`

**Comment by arunagw** [comment]
Is it because of newline is a whitespace?

**Comment by ihemantd** [comment]
xcode -> preferences -> text editing  has two tabs, Editing and Indentation.
I think i might be missing some settings.

File: ConferencesApp/ConferenceDataStore.swift [hunk 1]
```diff
@@ -88,6 +90,51 @@ class ConferenceDataStore {
...
+           
+                print("Updated calenderEventID")
+                debugPrint(calEvent)
+            }
+        } catch {
+                print("Could not save calendar event ID in database.")
+        }
+    }
+    
+    func deleteCalendarEventData(conf:Conference) -> Void {
```
Comment (raysrashmi) [inline ConferencesApp/ConferenceDataStore.swift]: Same here space after colon 
```conf: Conference```

Hound is suggesting the same

File: ConferencesApp/ConferenceDataStore.swift [hunk 2]
```diff
@@ -88,6 +90,51 @@ class ConferenceDataStore {
...
+                try realm.write {
+                     realm.delete(eventData)
+                }
+            }
+        } catch {
+            print("No calendar event for this conference.")
+        }
+    }
+    
+    func calendarEventIDFor(conf:Conference) -> String {
```
Comment (raysrashmi) [inline ConferencesApp/ConferenceDataStore.swift]: Same here :)

File: ConferencesApp/ConferenceTableViewController.swift [hunk 3]
```diff
@@ -115,12 +128,107 @@ class ConferenceTableViewController: UITableViewController, MKMapViewDelegate {
...
+    }
+    
+    func addCalendarEventtoStore(eventStore: EKEventStore) {
+        let event: EKEvent = EKEvent(eventStore: eventStore)
+        event.title = self.conferenceName
+        event.startDate = self.startDate
+        event.endDate = self.endDate
+        event.location = self.locationLabel.text
+        event.allDay = true
+        // event.notes = "Note for conference"
```
Comment (raysrashmi) [inline ConferencesApp/ConferenceTableViewController.swift]: If we are not using this we can remove this

File: ConferencesApp/ConferenceTableViewController.swift [hunk 4]
```diff
@@ -115,12 +128,107 @@ class ConferenceTableViewController: UITableViewController, MKMapViewDelegate {
...
+                    }
+
+                } else { // Event Already present in calendar.
+                    dispatch_async(dispatch_get_main_queue()) {
+                        self.deleteEventFromStore(eventStore, eventID:self.calendarEventID)
+                    }
+                }
+            }
+        })
+
```
Comment (raysrashmi) [inline ConferencesApp/ConferenceTableViewController.swift]: Can remove this extra line

File: ConferencesApp/ConferenceTableViewController.swift [hunk 5]
```diff
@@ -47,7 +52,15 @@ class ConferenceTableViewController: UITableViewController, MKMapViewDelegate {
         proposalLabel.text = conference.cfp_text
         dateLabel.text = conference.when
         locationLabel.text = conference.location
-
+        
+        conferenceName = conference.name
+        startDate =  conference.startDate
+        endDate = conference.endDate
+        calendarEventID = conferenceDataStore?.calendarEventIDFor(conference)
+        debugPrint(calendarEventID)
```
Comment (raysrashmi) [inline ConferencesApp/ConferenceTableViewController.swift]: Shouldn't we need to remove this?

File: ConferencesApp/ConferenceDataStore.swift [hunk 6]
```diff
@@ -88,6 +90,51 @@ class ConferenceDataStore {
         }
     }
     
+    func updateCalendarEventIdentifier( conf:Conference, eventId:String) {
```
Comment (raysrashmi) [inline ConferencesApp/ConferenceDataStore.swift]: Can you remove space before ```conf```?
It good be like ```(conf: Conference, eventId: String)```

File: ConferencesApp/Base.lproj/Main.storyboard [hunk 7]
```diff
@@ -1,5 +1,4 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="8191" systemVersion="15A284" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" initialViewController="K8A-N7-rbY">
```
Comment (raysrashmi) [inline ConferencesApp/Base.lproj/Main.storyboard]: I am getting error. you need to put this line back.

**Comment by raysrashmi** [comment]
@ihemantd Sorry to reply so late. I am using same 7.1.1 xcode version. thats strange you get these errors

**Comment by ihemantd** [comment]
@raysrashmi no problem.

**Comment by raysrashmi** [comment]
@ihemantd Hey, howz it going? Do you need any help on this? Actually we are planning to release a new version soon. so let us know if you need help from us

**Comment by ihemantd** [comment]
This PR has become too confusing for me with all the violation/commits/conflicts.
Can I submit fresh PR discarding this one. It Would be easy for me.

**Comment by raysrashmi** [comment]
@ihemantd Yup totally fine with us. Create new PR

**Pull request closed without merging**